### DISCUSSION TITLE
typescript type definitions?

### DISCUSSION THREAD

**Discussion by dsyang** [alpinejs/alpine#1033] [1. Help]
`npm install --save-dev @types/alpinejs` can't find any type definitions for this library.  Has anyone tried creating type definitions?

**Comment by HugoDF** [comment]
@dsyang you tend not to need to interact with Alpine, since it auto-inits on load, you just need to add `x-data` and whatever other directives and get coding

**Comment by dsyang** [comment]
@HugoDF  thanks for the response! I was trying to use [Alpinejs with Phoenix LiveView](https://underjord.io/getting-started-with-petal.html) by using `Alpine.clone` like so:
```js
// .. after the app.scss import
import Alpine from "alpinejs";
// .. in the let liveSocket configuration, after params
dom: {
    onBeforeElUpdated(from, to) {
        if (from.__x) {
        Alpine.clone(from.__x, to);
        }
    },
},
```
Trying to do that in typescript is what prompted me to look for type definitions.  There's a bunch of ways I can work around it though and, if I understand your comment, I'm likely never to have to do anything like this again so it's likely this isn't a big issue.

**Reply by HugoDF** [reply]
Ah right

This is an integration with a 3rd party library use case, yeah it's not the usual way people use the library 😂
